kbase-extension/static/kbase/js/widgets/appWidgets2/input/newObjectInput.js

```diff
@@ -80,49 +86,65 @@ define([
         }
 
         /*
-         *
          * Text fields can occur in multiples.
          * We have a choice, treat single-text fields as a own widget
          * or as a special case of multiple-entry --
          * with a min-items of 1 and max-items of 1.
-         *
-         *
          */
 
         function validate() {
-            return Promise.try(function() {
-                    if (!options.enabled) {
-                        return {
-                            isValid: true,
-                            validated: false,
-                            diagnosis: 'disabled'
-                        };
-                    }
-
-                    var rawValue = getInputValue(),
-                        validationOptions = {
-                            required: spec.data.constraints.required,
-                            shouldNotExist: true,
-                            workspaceId: workspaceId,
-                            types: spec.data.constraints.types,
-                            authToken: runtime.authToken(),
-                            workspaceServiceUrl: runtime.config('services.workspace.url')
-                        };
-
-                    return Validation.validateWorkspaceObjectName(rawValue, validationOptions);
-                })
-                .then(function(validationResult) {
-                    // TODO: should pass the validation object through untouched...
-                    return validationResult;
-                    //                    return {
-                    //                        isValid: validationResult.isValid,
-                    //                        validated: true,
-                    //                        diagnosis: validationResult.diagnosis,
-                    //                        errorMessage: validationResult.errorMessage,
-                    //                        shortMessage: validationResult.shortMessage,
-                    //                        value: validationResult.parsedValue
-                    //                    };
+            if (!options.enabled) {
+                return Promise.try(() => {
+                    return {
+                        isValid: true,
+                        validated: false,
+                        diagnosis: 'disabled'
+                    };
                 });
+            }
+            else {
+                var rawValue = getInputValue();
+                return validateUniqueOutput(rawValue)
+                    .then((isUnique) => {
+                        if (!isUnique) {
+                            return Promise.try(() => {
```

Comment:
this doesn't need to return a promise, just the value, since it is already wrapped in then, which itself always returns a promise.